staging/src/k8s.io/apiserver/pkg/server/filters/priority-and-fairness_test.go

```diff
@@ -333,6 +345,254 @@ func TestApfCancelWaitRequest(t *testing.T) {
 	})
 }
 
+func TestPriorityAndFairnessWithPanicRecoverAndTimeoutFilter(t *testing.T) {
+	fcmetrics.Register()
+
+	t.Run("priority level concurrency is set to 1, request handler panics, next request should not be rejected", func(t *testing.T) {
+		const (
+			requestTimeout = time.Minute
+			userName       = "alice"
+			fsName         = "test-fs"
+			plName         = "test-pl"
+			plConcurrency  = 1
+		)
+
+		serverConcurrency := getServerConcurrency(plConcurrency)
+		objects := newConfiguration(fsName, plName, userName, flowcontrol.LimitResponseTypeReject, plConcurrency)
+		clientset := newClientset(t, objects...)
+		factory := informers.NewSharedInformerFactory(clientset, time.Second)
+		controller := utilflowcontrol.New(factory, clientset.FlowcontrolV1beta1(), serverConcurrency, requestTimeout/4)
+
+		stopCh, stoppedCh := make(chan struct{}), make(chan struct{})
+		factory.Start(stopCh)
+
+		// wait for the informer cache to sync.
+		timeout, cancel := context.WithTimeout(context.TODO(), 2*time.Second)
+		defer cancel()
+		cacheSyncDone := factory.WaitForCacheSync(timeout.Done())
+		if names := check(cacheSyncDone); len(names) > 0 {
+			t.Fatalf("WaitForCacheSync did not successfully complete, resources=%#v", names)
+		}
+
+		var controllerErr error
+		go func() {
+			defer close(stoppedCh)
+			controllerErr = controller.Run(stopCh)
+		}()
+
+		// make sure that apf controller syncs the priority level configuration object we are using in this test.
+		// read the metrics and ensure the concurrency limit for our priority level is set to the expected value.
+		pollErr := wait.PollImmediate(100*time.Millisecond, 1*time.Second, func() (done bool, err error) {
+			if err := gaugeValueMatch("apiserver_flowcontrol_request_concurrency_limit", map[string]string{"priority_level": "test-pl"}, plConcurrency); err != nil {
+				t.Logf("polling retry - error: %s", err)
+				return false, nil
+			}
+			return true, nil
+		})
+		if pollErr != nil {
+			t.Fatalf("expected the apf controller to sync the priotity level configuration object: %s", "test-pl")
+		}
+
+		var executed bool
+		// we will raise a panic for the first request.
+		firstRequestPathPanic := "/request/panic"
+		requestHandler := http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			executed = true
+			expectMatchingAPFHeaders(t, w, fsName, plName)
+
+			if r.URL.Path == firstRequestPathPanic {
+				panic(fmt.Errorf("request handler panic'd - %#v", r.RequestURI))
```

Comment:
Let's make it clear to anyone reading the output that this is normal.  How about something like

> panic(fmt.Errorf("request handler panic'd as designed - %#v", r.RequestURI))

?